src/main/webapp/resources/js/pages/projects/redux/samplesSlice.js

```diff
@@ -3,21 +3,49 @@ import {
   createAsyncThunk,
   createReducer,
 } from "@reduxjs/toolkit";
-import { getProjectIdFromUrl } from "../../../utilities/url-utilities";
+import {
+  getProjectIdFromUrl,
+  setBaseUrl,
+} from "../../../utilities/url-utilities";
 import { INITIAL_TABLE_STATE } from "../samples/services/constants";
 import { getMinimalSampleDetailsForFilteredProject } from "../../../apis/projects/samples";
 import { putSampleInCart } from "../../../apis/cart/cart";
 import { downloadPost } from "../../../utilities/file-utilities";
 import { formatFilterBySampleNames } from "../../../utilities/table-utilities";
-import { setBaseUrl } from "../../../utilities/url-utilities";
+import isEqual from "lodash/isEqual";
 
-const updateTable = createAction("samples/table/update");
 const reloadTable = createAction("samples/table/reload");
 const addSelectedSample = createAction("samples/table/selected/add");
 const removeSelectedSample = createAction("samples/table/selected/remove");
 const clearSelectedSamples = createAction("samples/table/selected/clear");
 const clearFilterByFile = createAction("samples/table/clearFilterByFile");
 
+/**
+ * Updates the state of the table filters and search, which triggers
+ * the re-render of the samples table.
+ * @type {AsyncThunk<unknown, void, {}>}
+ */
+const updateTable = createAsyncThunk(
+  "samples/table/update",
+  async (values, { getState }) => {
+    const {
+      samples: { options, selected, selectedCount },
+    } = getState();
+    console.log("SEARCH", { state: options.search, values: values.search });
```

Comment:
Remove the logging statements here and below

Reply:
Removed in d73f8bb385198793f63345d399f52a4226e6be79